src/loctool/PendoXliffFileType.ts

```diff
@@ -0,0 +1,347 @@
+        const localeMap = this.project.settings?.localeMap as unknown;
+
+        // if there is a locale mapping, use it
+        if (
+            localeMap &&
+            typeof localeMap === "object" &&
+            loctoolLocale in localeMap &&
+            // @ts-expect-error -- undocumented
+            typeof localeMap[loctoolLocale] === "string" &&
+            // @ts-expect-error -- undocumented
+            localeMap[loctoolLocale].length > 0
+        ) {
+            // @ts-expect-error -- undocumented
+            return localeMap[loctoolLocale];
+        }
+
+        return loctoolLocale;
+    }
+
+    /**
+     * Wraps translation units in loctool's ResourceString objects.
+     */
+    private createTranslationSet(units: TranslationUnit[]) {
+        // convert to loctool resources
+        const resources = units.map((unit) =>
+            this.loctoolAPI.newResource({
+                resType: "string",
+                key: unit.key,
+                sourceLocale: unit.sourceLocale,
+                source: unit.source,
+                comment: unit.comment,
+                // not sure if this is used by loctool anywhere,
+                // but it's required per the Resource interface definition
+                project: this.project.getProjectId(),
+            }),
+        );
+
+        // wrap in a translation set
+        const translationSet = this.loctoolAPI.newTranslationSet<ResourceString>(this.project.getSourceLocale());
+        translationSet.addAll(resources);
+
+        return translationSet;
+    }
+
+    getExtracted(): TranslationSet {
+        // this should ouput a translationset with merged resources from all files
+        // which fit this filetype;
+        // note: this means that the instance of the filetype class has to keep track of all files it
+        // has processed (i.e. each {@link newFile} call should add the file to a list)
+
+        const translationSet = this.loctoolAPI.newTranslationSet(this.sourceLocale);
+
+        for (const path in this.files) {
+            const file = this.files[path];
+            translationSet.addSet(file.getTranslationSet());
+        }
+
+        return translationSet;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars -- noop, see comment in method
+    addSet(set: TranslationSet): void {
+        // no-op
+        // it's not clear why would the filetype need to add a set of translations to itself
+        // given that per the getExtracted method, sets should come from the files
+    }
+
+    getNew(): TranslationSet {
+        // not sure how this differs from getExtracted
+        return this.getExtracted();
+    }
+
+    getPseudo(): TranslationSet {
+        return this.loctoolAPI.newTranslationSet(this.sourceLocale);
+    }
+
+    getResourceTypes() {
+        // per https://github.com/iLib-js/loctool/blob/285401359f923c1be11e7329b549ed11b4099637/lib/Project.js#L235-L244:
```

Comment:
This method is not needed for this plugin. It is there to map the datatype attribute in the xliff to a subclass of Resource that is different than the default. (ie. if you want "string" to map to a ContextResourceString instance vs. a regular ResourceString instance) The default is fine in this case.

Reply:
I'm getting the following error without it
```
16:04:13 INFO loctool.loctool: -------------------------------------------------------------------------------------------
16:04:13 ERROR loctool.loctool: caught exception: TypeError: type.getResourceTypes is not a function
16:04:13 ERROR loctool.loctool: TypeError: type.getResourceTypes is not a function
    at /Users/wwawrzenczak/Development/iLib-js/loctool/lib/Project.js:238:68
    at Array.forEach (<anonymous>)
    at CustomProject.<anonymous> (/Users/wwawrzenczak/Development/iLib-js/loctool/lib/Project.js:234:24)
    at Project.initFileTypes (/Users/wwawrzenczak/Development/iLib-js/loctool/lib/Project.js:214:9)
    at Project.initFileTypes (/Users/wwawrzenczak/Development/iLib-js/loctool/lib/Project.js:211:18)
    at Project.init (/Users/wwawrzenczak/Development/iLib-js/loctool/lib/Project.js:230:10)
    at processNextProject (/Users/wwawrzenczak/Development/iLib-js/loctool/loctool.js:553:17)
    at Object.<anonymous> (/Users/wwawrzenczak/Development/iLib-js/loctool/loctool.js:727:9)
    at Module._compile (node:internal/modules/cjs/loader:1364:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1422:10)
16:04:13 INFO loctool.loctool: Done
```
As described in the comments, I looked at the following code https://github.com/iLib-js/loctool/blob/285401359f923c1be11e7329b549ed11b4099637/lib/Project.js#L235-L244
and it seems that either of those is needed ```ts
registerDataTypes(): void
getResourceTypes(): Record<Datatype, ResourceClassName>
```